internal/pkg/consensus/election.go

```diff
@@ -12,9 +12,39 @@ import (
 	"github.com/filecoin-project/go-filecoin/internal/pkg/vm/address"
 )
 
-// CompareElectionPower return true if the input electionProof is below the
+// ElectionMachine generates and validates PoSt partial tickets and PoSt
+// proofs.
+type ElectionMachine struct{}
+
+// func (em ElectionMachine) GeneratePoStRandomness(ticket block.Ticket, candidateAddr address.Address, signer types.Signer, nullBlockCount uint64) (block.VRFPi, error) {
```

Comment:
This is "Dragons".  I can add that explicitly or remove this commented out stuff right for this PR if preferred.  However this pseudo code is helpful to me and will be replaced with the real deal in a day or two.